mvlearn/decomposition/groupica.py

```diff
@@ -0,0 +1,293 @@
+            )
+            sources = sources.T
+        if K is not None:
+            self.components_ = np.dot(W, K)
+        else:
+            self.components_ = W
+        self.mixing_ = linalg.pinv(self.components_)
+        # Compute individual unmixing matrices by least-squares
+        self.individual_mixing_ = []
+        self.individual_components_ = []
+        sources_pinv = linalg.pinv(sources)
+        for X, mean in zip(Xs, self.means_):
+            lstq_solution = np.dot(sources_pinv, X - mean)
+            self.individual_components_.append(linalg.pinv(lstq_solution).T)
+            self.individual_mixing_.append(lstq_solution.T)
+        self.n_components_ = gpca.n_components_
+        self.n_features_ = gpca.n_features_
+        self.n_samples_ = gpca.n_samples_
+        self.n_subjects_ = gpca.n_subjects_
+        return sources
+
+    def fit_transform(self, Xs, y=None):
+        """
+        Fit  to the data and transform the data
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : array of shape (n_samples, n_components)
+            The transformed data
+        """
+        sources = self._fit(Xs, y)
+        if self.multiple_outputs:
+            return self.transform(Xs)
+        else:
+            return sources
+
+    def fit(self, Xs, y=None):
+        r"""Fit the model with Xs.
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : None
+            Ignored variable.
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        self._fit(Xs, y)
+        return self
+
+    def transform(self, Xs, y=None):
+        r"""
+        A method to fit model to multiview data.
+
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : array of shape (n_samples, n_components)
+            The transformed data
+        """
+        Xs = check_Xs(Xs, copy=True)
+        if self.multiple_outputs:
+            return [
+                np.dot(X - mean, W.T)
+                for W, X, mean in (
+                    zip(self.individual_components_, Xs, self.means_)
+                )
+            ]
+        else:
+            X = self.grouppca_.transform(Xs)
+            return np.dot(X, self.components_.T)
+
+    def inverse_transform(self, X_transformed):
+        r"""
+        A method to recover multiview data from transformed data
+        """
```

Comment:
Unless this is only supposed by be used privately (in which case add an underscore), it should have Parameters and Returns docs